### PR TITLE
Feat/keystore shim

### PR DESCRIPTION
**Pull request by reenigneEsrever92**
Maybe this could serve as a discussion basis for: https://github.com/Azure/azure-sdk-for-rust/issues/803

### CODE REVIEW

File: sdk/keyvault/src/keyvault_client.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+use azure_core::auth::TokenCredential;
+use azure_svc_keyvault::{
+    create_key::Response,
+    models::{
+        key_create_parameters::{Crv, Kty},
+        KeyAttributes, KeyCreateParameters, KeyReleasePolicy,
+    },
+};
+
+pub struct KeyvaultClient {
```
Comment (heaths) [inline sdk/keyvault/src/keyvault_client.rs:12]: We already have Key Vault clients for keys, secrets, and certs. Why add another one that isn't scoped like all our other languages? We want our SDKs to be idiomatic, but still similarly designed.

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/keyvault/src/keyvault_client.rs (hunk 1 above)
Comment (cataggar) [reply sdk/keyvault/src/keyvault_client.rs:12]: This looks to be for #803, where we look to migrate our SDKs to be on top of the generated service crates.

Comment (heaths) [reply sdk/keyvault/src/keyvault_client.rs:12]: Why is it under `sdk/keyvault` then? The structures of SDKs should remain idiomatic but consistent with all other languages' Azure SDKs.

**Comment by cataggar** [comment]
Thanks for this prototype. Yes, we want to modifying the SDK `azure_security_keyvault` to use the generated `azure_svc_keyvault` crate.

**Pull request closed without merging**